fipy/tools/numerix.py

```diff
@@ -1007,6 +1007,48 @@ def in1d(ar1, ar2, assume_unique=False):
         else:
             return flag[indx][rev_idx]
 
+def lists_to_masked(x):
+    """Convert list of lists of different length to Numpy masked array
+
+    See https://stackoverflow.com/questions/38619143/convert-python-sequence-to-numpy-array-filling-missing-values
+
+    >>> print(lists_to_masked([[0], [0, 1], [0, 1, 2]]))
+    [[0 0 0]
+     [-- 1 1]
+     [-- -- 2]]
+
+    After https://stackoverflow.com/a/59686318/2019542
+    """
+    import itertools
+
+    return MA.masked_values(list(itertools.zip_longest(*x, fillvalue=-1)), -1)
+
+
+def invert_indices(arr, axis=-1):
+    """Invert an index array
+
+    Given an array of indices, return the locations in the array along
+    `axis` of each index.
+
+    >>> a = numpy.array([[0, 2], [1, 3], [0, 3], [3, 4]])
```

Comment:
Should be NUMERIX here, but maybe better to just remove NUMERIX altogether and use numpy

Reply:
You know that as soon as we remove `NUMERIX` that they'll release some reason we'll want it back. It ain't hurtin' nuthin'.